samples/Java/azure-servicebus/QueueClientQuickstart/src/main/java/com/microsoft/azure/servicebus/samples/queueclientquickstart/QueueClientQuickstart.java

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package com.microsoft.azure.servicebus.samples.queueclientquickstart;
+
+import com.microsoft.azure.servicebus.*;
+import com.microsoft.azure.servicebus.primitives.ConnectionStringBuilder;
+import org.apache.commons.cli.*;
+import org.apache.log4j.*;
+
+import java.time.Duration;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class QueueClientQuickstart {
+    // Connection String for the namespace can be obtained from the Azure portal under the
+    // 'Shared Access policies' section.
+
+    // connection string can be initialized via environment variable
+    private static String connectionString = System.getenv("SB_SAMPLES_CONNECTIONSTRING");
+    // queue name can be initialiyzed via environment variable
+    private static String queueName = System.getenv("SB_SAMPLES_QUEUENAME");
+    // queue client instance
+    private static IQueueClient queueClient;
+    // message send/receive counters
+    private static int totalToSend = 100;
+    private static AtomicInteger totalReceived = new AtomicInteger(0);
+    // log4j logger 
+    private static Logger logger = Logger.getRootLogger();
+
+    public static void main(String[] args) throws Exception {
+        // Parse and evaluate command line. Exit when there's not
+        // enough configuration information in environment and to 
+        // command line to continue
+        if (!parseCommandLine(args)) {
+            return;
+        }
+
+        logger.info("Starting BasicSendReceiveWithQueueClient sample");
+
+        // Create a QueueClient instance using the connection string builder
+        logger.info("Create queue client.");
+        queueClient = new QueueClient(new ConnectionStringBuilder(connectionString, queueName), ReceiveMode.PEEKLOCK);
```

Comment:
>ReceiveMode.PEEKLOCK [](start = 96, length = 20)

Given that this is the first sample in this tutorial/sample, maybe a small description of what `PeekLock` is might be easier for users. Difference between `PeekLock` and `ReceiveAndDelete` and probably what is `AutoComplete`

Reply:
We're currently staging a big update to the conceptual Service Bus docs. I actually want to link to the concepts from the sample comments, probably with shortened URLs. I'll add a comment.